feat: Codacy uses configuration files even if ignored IO-539 (#1779)
codacy/docs: docs/repositories-configure/configuring-code-patterns.md

-   Codacy will use the version of the configuration file **in the branch being analyzed**. For example, if you open a pull request that includes changes to the configuration file, the analysis results will take those changes into account.
-   If Codacy analyzes a branch that doesn't include the configuration file, Codacy reverts to using the code patterns configured for the tool before you selected the option **Configuration file** on the Code patterns page.

> -   Codacy still uses the configuration file even if you ignore it using either [the Codacy UI](ignoring-files.md) or the [Codacy configuration file](codacy-configuration-file.md).

Author: @MiguelMarcelino, I found the issue [IO-539](https://codacy.atlassian.net/browse/IO-539) and learned about the described behavior.

Would you say that the product "works as designed" or are there reasons for changing how Codacy works so that customers can ignore tool configuration files to force Codacy to use the UI settings for the tools? Or, put another way, should we document this behavior now or wait until the results of your investigation on [IO-711](https://codacy.atlassian.net/browse/IO-711)?

Reviewer: I would say that after my investigation in [IO-539](https://codacy.atlassian.net/browse/IO-539) that this is the intended behaviour. So I would say that this is correct and that we can add this to the documentation. The investigation on [IO-711](https://codacy.atlassian.net/browse/IO-711) will reveal which tools are dependent on this behaviour, I believe.

Author: Thanks for your input, @MiguelMarcelino! :bow: I'm going to merge this change now and we can always reassess later in the scope of [IO-711](https://codacy.atlassian.net/browse/IO-711). :rocket:

> -   Codacy still uses the configuration file even if you ignore it using either [the Codacy UI](ignoring-files.md) or the [Codacy configuration file](codacy-configuration-file.md).

Reviewer: Minor suggestion, altough not blocking
```suggestion
-   Codacy will use tool configuration files even if you ignore them through [Codacy's UI](ignoring-files.md) or in a [Codacy configuration file](codacy-configuration-file.md).
```

Author: Thank you @MiguelMarcelino! Here's one more edit building on top of your suggestion to avoid the apostrophe on "Codacy's" and also improve the link text. :+1: 

```suggestion
-   Codacy will use tool configuration files even if you ignore them [using the Codacy UI](ignoring-files.md) or [in a Codacy configuration file](codacy-configuration-file.md).
```
